packages/docusaurus-theme-classic/src/theme/Navbar/index.js

```diff
@@ -29,7 +29,8 @@ function NavLink({
   ...props
 }) {
   const toUrl = useBaseUrl(to);
-  const activeBaseUrl = useBaseUrl(activeBasePath);
+  const activeBaseUrl = useBaseUrl(activeBasePath.toString());
+  const activeBasePathIsRegex = typeof activeBasePath === 'object';
```

Comment:
> typeof activeBasePath === 'object';

Why not `activeBasePath instanceof RegExp` or something like that?

Reply:
So, there's a bug. For some reason, none of the methods I've used to check if `activeBaseUrl` is a Regular Expression work. It's almost like React Context breaks RegExps.

Reply:
Instead, `activeBasePath` shows up (when it should be a RegExp) as an empty object.

Reply:
But currently it looks rather unreliable, we need to find out what exactly is the matter.

Reply:
I almost think there's a bug, where React Context can't handle Regular Expressions correctly.

Reply:
I'll try to investigate this issue.